pyfr/mixins/writer.py

```diff
@@ -0,0 +1,34 @@
+# -*- coding: utf-8 -*-
+
+import re
+import os
+
+class WriterMixin:
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    def _get_output_path(self, tcurr, basename=None):
+        if basename is None:
+            basename = self.basename
+
+        # Substitute {t} and {n} for the current time and output number
+        fname = basename.format(t=tcurr, n=self.nout[basename])
+        return os.path.join(self.basedir, fname)
+
+    def _restore_nout(self, basename=None):
```

Comment:
Seems a bit convoluted as a mixin.  See if it is cleaner to write a somewhat generic _restore_nout function which can go in until.py.